### PR TITLE
Added functionality to deal with network errors in download_hydat()

### LINKED ISSUE #4: Building a vignette
When a vignette is built using `devtools::build_vignette()` all built files are moved over to [here](https://github.com/bcgov/tidyhydat/tree/master/inst/doc). Normally those files are part of gitignore. However travis  was throwing a warning if built files weren't in the repo. So I built the vignette and remove isnt/doc from gitignore. 

This has the unintended effect of removing `tidyhydat.md` which was produced using [this](https://github.com/bcgov/bcgovr/issues/11) method. 

Here is what I want. Build the [vignette](https://github.com/bcgov/tidyhydat/blob/master/vignettes/tidyhydat.Rmd) so that a plain markdonw file is generated that includes all the references from the bibtex file and figure out a ways so that TRAVIS doesn't expect the inst/doc directory.

### LINKED ISSUE #101: Consider adding a link to the WSC website to download HYDAT database manually
For those of us behind corporate proxies, the generic `download_hydat()` fails unless you're knowledgeable enough to set up your system proxy information. It could be argued that this is outside the scope of `tidyhydat` but I mentioning/linking the source on the [WSC](https://wateroffice.ec.gc.ca/index_e.html) probably doesn't hurt. 

Typical proxy error below:

```
download_hydat()
Downloading HYDAT will take ~10 minutes.
This will remove any older versions of HYDAT
Is that okay?
--------------------------------------------------------------------------------

1: Yes
2: No

Selection: 1
* Downloading HYDAT.sqlite3 to C:\Users\whaleyry\AppData\Local\tidyhydat\tidyhydat
Error in curl::curl_fetch_memory(url, handle = handle) : 
  Timeout was reached: Send failure: Connection was aborted

```

### PR DESCRIPTION
**Pull request by rywhale**
``download_hydat()`` fails behind firewalls when the user's proxy info isn't set. This accounts for that possibility and prompts users to download HYDAT manually via a browser. 

## Description

Created a function (``network_check()``) that checks for timeout errors when running ``download_hydat()``. If timeout occurs, the user is given a direct link to HYDAT and prompted to download the database manually. User is also given the path where HYDAT should be placed after downloading. 

This has the added benefit of being faster than downloading HYDAT through R though it complicates the process for new users. 

Also added a test to ``test_utils.R`` to confirm ``network_check()`` functions as intended. 

## Related Issue

Fix #101 

## Example
```
tidyhydat::download_hydat()
Downloading HYDAT will take ~10 minutes.
This will remove any older versions of HYDAT
Is that okay?
--------------------------------------------------------------------------------

1: Yes
2: No

Selection: 1
* Downloading HYDAT.sqlite3 to C:\Users\whaleyry\AppData\Local\tidyhydat\tidyhydat
Error: Could not connect to HYDAT source. Check your connection settings.
                           Try downloading HYDAT_sqlite3 from this url: 
                           [http://collaboration.cmc.ec.gc.ca/cmc/hydrometrics/www/]
                           and unzipping the saved file to this directory:C:\Users\whaleyry\AppData\Local\tidyhydat\tidyhydat
````

### CODE REVIEW

File: tests/testthat/test_utils.R [hunk 1]
```diff
@@ -24,3 +24,15 @@ test_that("hy_version returns a dataframe and works",{
   expect_silent(hy_version(hydat_path = hy_test_db()))
 })
 
+test_that("downloading hydat fails behind a proxy server with informative error message",{
+  httr::set_config(httr::use_proxy(url = "http://google.com", port = 1234), override = TRUE)
+  base_url <-
+    "http://collaboration.cmc.ec.gc.ca/cmc/hydrometrics/www/"
+  expect_error(network_check(base_url), messsage = paste0("Error: Could not connect to HYDAT source.
```
Comment (boshek) [inline tests/testthat/test_utils.R]: typo in `messsage` . Replace with message I think.

File: tests/testthat/test_utils.R [hunk 2]
```diff
@@ -24,3 +24,15 @@ test_that("hy_version returns a dataframe and works",{
...
 
+test_that("downloading hydat fails behind a proxy server with informative error message",{
+  httr::set_config(httr::use_proxy(url = "http://google.com", port = 1234), override = TRUE)
+  base_url <-
+    "http://collaboration.cmc.ec.gc.ca/cmc/hydrometrics/www/"
+  expect_error(network_check(base_url), messsage = paste0("Error: Could not connect to HYDAT source. 
+                                                    Check your connection settings.
+                                                    Try downloading HYDAT_sqlite3 from this url: 
+                                                    [http://collaboration.cmc.ec.gc.ca/cmc/hydrometrics/www/]
+                                                    and unzipping the saved file to this directory:",
```
Comment (boshek) [inline tests/testthat/test_utils.R]: Please add a space after the `:` so that the message prints nicer

File: R/utils.R [hunk 3]
```diff
@@ -181,10 +181,14 @@ ask <- function(...) {
...
 network_check <- function(url){
-  tryCatch(httr::GET(base_url),
+  tryCatch(httr::GET(url),
            error = function(e){
              if(grepl("Timeout was reached:", e$message))
-               stop("Could not connect to HYDAT source. Check your connection settings.", 
+               stop(paste0("Could not connect to HYDAT source. Check your connection settings.
+                           Try downloading HYDAT_sqlite3 from this url: 
+                           [http://collaboration.cmc.ec.gc.ca/cmc/hydrometrics/www/]
+                           and unzipping the saved file to this directory:",
```
Comment (boshek) [inline R/utils.R]: Please add a space after the `:` so that the message prints nicer

File: R/utils.R [hunk 4]
```diff
@@ -181,10 +181,14 @@ ask <- function(...) {
 # Deal with proxy-related connection issues
```
Comment (boshek) [inline R/utils.R]: Just a little more detail. Replace "Deal..." with something like "Catch network timeout error generated when dealing with proxy-related connection issues and fail with a information error message of where to download HYDAT"

**Comment by rywhale** [comment]
I think I managed to address everything you mentioned. 

Output from R CMD check pasted below. The only warning came from 'qdpf' not being available to check for size reductions of PDFs. Seems it isn't available for my version of R. Happy to downgrade and run this again if you think it's important.

```
* using log directory 'C:/Users/Diogenes/AppData/Local/Temp/RtmpaopHx2/tidyhydat.Rcheck'
* using R version 3.5.0 (2018-04-23)
* using platform: x86_64-w64-mingw32 (64-bit)
* using session charset: ISO8859-1
* using options '--no-manual --as-cran'
* checking for file 'tidyhydat/DESCRIPTION' ... OK
* this is package 'tidyhydat' version '0.3.4.9000'
* package encoding: UTF-8
* checking package namespace information ... OK
* checking package dependencies ... NOTE
Package suggested but not available for checking: 'covr'
* checking if this is a source package ... OK
* checking if there is a namespace ... OK
* checking for .dll and .exe files ... OK
* checking for hidden files and directories ... OK
* checking for portable file names ... OK
* checking serialization versions ... OK
* checking whether package 'tidyhydat' can be installed ... OK
* checking package directory ... OK
* checking 'build' directory ... OK
* checking DESCRIPTION meta-information ... OK
* checking top-level files ... OK
* checking for left-over files ... OK
* checking index information ... OK
* checking package subdirectories ... OK
* checking R files for non-ASCII characters ... OK
* checking R files for syntax errors ... OK
* checking whether the package can be loaded ... OK
* checking whether the package can be loaded with stated dependencies ... OK
* checking whether the package can be unloaded cleanly ... OK
* checking whether the namespace can be loaded with stated dependencies ... OK
* checking whether the namespace can be unloaded cleanly ... OK
* checking loading without being on the library search path ... OK
* checking dependencies in R code ... OK
* checking S3 generic/method consistency ... OK
* checking replacement functions ... OK
* checking foreign function calls ... OK
* checking R code for possible problems ... OK
* checking Rd files ... OK
* checking Rd metadata ... OK
* checking Rd line widths ... OK
* checking Rd cross-references ... OK
* checking for missing documentation entries ... OK
* checking for code/documentation mismatches ... OK
* checking Rd \usage sections ... OK
* checking Rd contents ... OK
* checking for unstated dependencies in examples ... OK
* checking contents of 'data' directory ... OK
* checking data for non-ASCII characters ... OK
* checking data for ASCII and uncompressed saves ... OK
 WARNING
'qpdf' is needed for checks on size reduction of PDFs
* checking installed files from 'inst/doc' ... OK
* checking files in 'vignettes' ... OK
* checking examples ... OK
* checking for unstated dependencies in 'tests' ... OK
* checking tests ... OK
  Running 'testthat.R' [43s]
* checking for unstated dependencies in vignettes ... OK
* checking package vignettes in 'inst/doc' ... OK
* checking re-building of vignette outputs ... OK
* DONE
Status: 1 WARNING, 1 NOTE
```

**Comment by rywhale** [comment]
@boshek Small unrelated thing, looks like the package version is still 0.3.4.9000 in the `DESCRIPTION` but we're at 0.3.6 in the `NEWS` file, dunno if this is standard practice or a small error. .

**Comment by boshek** [comment]
> Small unrelated thing, looks like the package version is still 0.3.4.9000 in the DESCRIPTION but we're at 0.3.6 in the NEWS file, dunno if this is standard practice or a small error.

Here is the way I handle this type of thing: Increment the version in the description to *.9000 to let users know (and session info type things) that this is the dev version of the package. But in the NEWS file (so I don't have to change it multiple times) I create a heading for the version I am working towards. So it should actually be 3.5.0. I am working on some other stuff right now so I don't want to merge this quite yet. Either I'll change once I've merged or if you get around to it just add your news entry into the 0.3.5 heading.

**Pull request merged**